runner/jobs.go

```diff
@@ -139,3 +141,89 @@ func (j Job) RunHaddock3(cmd string) (string, error) {
 	return logF, nil
 
 }
+
+func (j Job) Run(version int, cmd string) (string, error) {
+
+	var logF string
+	var err error
+
+	switch version {
+	case 2:
+		errSetup2 := j.SetupHaddock24(cmd)
+		if errSetup2 != nil {
+			err := errors.New("Failed to setup HADDOCK: " + errSetup2.Error())
+			return logF, err
+		}
+		logF, _ = j.RunHaddock24(cmd)
+		// if err != nil {
+		// 	err := errors.New("Failed to run HADDOCK: " + err.Error())
+		// 	return logF, err
+		// }
+	case 3:
+		logF, err = j.RunHaddock3(cmd)
+		if err != nil {
+			err := errors.New("Failed to run HADDOCK: " + err.Error())
+			return logF, err
+		}
+	}
+
+	return logF, nil
+
+}
+
+func (j *Job) GetStatus(version int) error {
+
+	var logF string
+	var positiveKeys []string
+	var negativeKeys []string
+
+	if version == 2 {
+		logF = filepath.Join(j.Path, "run1", "haddock.out")
+		positiveKeys = []string{"Finishing HADDOCK on:"}
+		negativeKeys = []string{"An error has occurred"}
+
+	} else if version == 3 {
+		logF = filepath.Join(j.Path, "run1", "log")
+		negativeKeys = []string{"ERROR"}
+		positiveKeys = []string{"This HADDOCK3 run took"}
```

Comment:
Each version has an array of positive and negative keys that are used to get the state